client/js/lounge.js

```diff
@@ -442,8 +442,11 @@ $(function() {
 				target,
 				data
 			]);
-
-		if (data.msg.self) {
+		var lastVisible = container.find("div:visible").last();
+		if (data.msg.self
+			|| lastVisible.hasClass("unread-marker")
+			|| (lastVisible.hasClass("date-marker")
+				&& lastVisible.prev().hasClass("unread-marker"))) {
```

Comment:
Would this prev() need visible check too? I understand why its necessary to check previous element, but this does not feel like performant code 😞